src/main/2.0/scala/com/holdenkarau/spark/testing/receiver/InputStreamTestingContext.scala

```diff
@@ -0,0 +1,65 @@
+package com.holdenkarau.spark.testing.receiver
+
+import org.apache.spark.SparkContext
+import org.apache.spark.streaming.dstream.DStream
+import org.apache.spark.streaming.{Duration, Seconds, StreamingContext}
+
+/**
+  * Sequences the execution of user-provided code that comprises one individual test of an
+  * [[InputStreamSuiteBase]] test suite. Each individual test will typically instantiate one instance of this class.
+  *
+  * @param sparkContext        - a  [[com.holdenkarau.spark.testing.SharedSparkContext]] instance, provided by
+  *                            [[com.holdenkarau.spark.testing.StreamingSuiteBase]], which will be used to create
+  *                            an instance of  a [[org.apache.spark.streaming.StreamingContext]] that will be used
+  *                            for a run of one individual test.
+  *
+  * @param dStreamCreationFunc - a user-provided code block that uses  the StreamingContext provided by this class
```

Comment:
Tonight I will look @ seeing how to make testDataGenerator function   take the output of the DStream creator function.